test/unit/common_pdfstream_tests.js

```diff
@@ -0,0 +1,98 @@
+/* Copyright 2024 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AbortException, isNodeJS } from "../../src/shared/util.js";
+import { getCrossOriginHostname, TestPdfsServer } from "./test_utils.js";
+
+// Common tests to verify behavior across implementations of the IPDFStream
+// interface:
+// - PDFNetworkStream by network_spec.js
+// - PDFFetchStream by fetch_stream_spec.js
+async function testCrossOriginRedirects({
+  PDFStreamClass,
+  redirectIfRange,
+  testRangeReader,
+}) {
+  const basicApiUrl = TestPdfsServer.resolveURL("basicapi.pdf").href;
+  const basicApiFileLength = 105779;
+
+  const rangeSize = 32768;
+  const stream = new PDFStreamClass({
+    url: getCrossOriginUrlWithRedirects(basicApiUrl, redirectIfRange),
+    length: basicApiFileLength,
+    rangeChunkSize: rangeSize,
+    disableStream: true,
+    disableRange: false,
+  });
+
+  const fullReader = stream.getFullReader();
+
+  await fullReader.headersReady;
+  // Sanity check: We can only test range requests if supported:
+  expect(fullReader.isRangeSupported).toEqual(true);
+
+  // When range requests are supported, the initial full request is aborted.
+  if (PDFStreamClass.name === "PDFNetworkStream") {
+    expect(stream._manager.pendingRequests).toEqual({});
+    expect(fullReader._done).toEqual(false);
+  } else if (PDFStreamClass.name === "PDFFetchStream") {
+    expect(fullReader._abortController.signal.aborted).toEqual(true);
```

Comment:
Reaching into the class-instances and checking various semi-private properties, that we may at some point want to convert to actually private fields, feels a little bit undesirable to me.

Reply:
This check is not critical. If we ever refactor this, we can either add a "public" interface to look up the state of the request, or simply drop the test logic here.

The current implementations abort when the server signals support for range requests, but I didn't see test coverage for that, so I added some minimal checks here.

If you prefer to not inspect internal members at the expense of test coverage, I can drop these checks from the test.

Reply:
I think that we should do *either* of the following, whatever is easiest here:
 - Just drop the checks of "internal" properties.
 - Add test-only helpers, protected by pre-processor checks [similar to here](https://github.com/mozilla/pdf.js/blob/22babd722fcdc6dddd9d77bf51a3eb39f58ff8b2/src/display/api.js#L802-L819), such that we can call one method regardless of what `IPDFStream` implementation we're testing.

Reply:
I dropped the checks, as they are not critical to this test. Here is the diff between the previous and current version of the patch:

```diff
diff --git a/test/unit/common_pdfstream_tests.js b/test/unit/common_pdfstream_tests.js
index f3efb4348..76ad87b4d 100644
--- a/test/unit/common_pdfstream_tests.js
+++ b/test/unit/common_pdfstream_tests.js
@@ -42,16 +42,8 @@ async function testCrossOriginRedirects({
   await fullReader.headersReady;
   // Sanity check: We can only test range requests if supported:
   expect(fullReader.isRangeSupported).toEqual(true);
-
-  // When range requests are supported, the initial full request is aborted.
-  if (PDFStreamClass.name === "PDFNetworkStream") {
-    expect(stream._manager.pendingRequests).toEqual({});
-    expect(fullReader._done).toEqual(false);
-  } else if (PDFStreamClass.name === "PDFFetchStream") {
-    expect(fullReader._abortController.signal.aborted).toEqual(true);
-  } else {
-    throw new Error(`Unexpected stream class: ${PDFStreamClass.name}`);
-  }
+  // ^ When range requests are supported, the initial full request is aborted,
+  // and we do not need to call fullReader.cancel().
 
   const rangeReader = stream.getRangeReader(
     basicApiFileLength - rangeSize,
```